src/com/goide/runconfig/GoBuildingRunner.java

```diff
@@ -79,46 +85,86 @@ public boolean canRun(@NotNull String executorId, @NotNull RunProfile profile) {
   @Override
   protected Promise<RunProfileStarter> prepare(@NotNull ExecutionEnvironment environment, @NotNull final RunProfileState state)
     throws ExecutionException {
-    final File outputFile = getOutputFile(environment, (GoApplicationRunningState)state);
     FileDocumentManager.getInstance().saveAllDocuments();
 
+    if (!(state instanceof GoRunningState)) {
+      throw new ExecutionException("Run state is not a valid GoRunningState");
+    }
+
     final AsyncPromise<RunProfileStarter> buildingPromise = new AsyncPromise<RunProfileStarter>();
     final GoHistoryProcessListener historyProcessListener = new GoHistoryProcessListener();
-    ((GoApplicationRunningState)state).createCommonExecutor()
-      .withParameters("build")
-      .withParameterString(((GoApplicationRunningState)state).getGoBuildParams())
-      .withParameters("-o", outputFile.getAbsolutePath())
-      .withParameters(((GoApplicationRunningState)state).isDebug() ? new String[]{"-gcflags", "-N -l"} : ArrayUtil.EMPTY_STRING_ARRAY)
-      .withParameters(((GoApplicationRunningState)state).getTarget())
+
+    String goToolCommand;
+    File outputFile;
+    String target;
+
+    if (state instanceof GoApplicationRunningState) {
```

Comment:
The particular state should know how to build itself, there is no need to do these instanceofs in generic builder